syncer/eth.go

```
@@ -113,3 +117,30 @@ func (es *EthSyncer) getNonce(client *ethclient.Client, account common.Address,
 	}
 	return nonce, nil
 }
+
+func (es *EthSyncer) getBalance(address string) (*big.Int, error) {
+	httpClient := newHTTPClient()
+	config := config.GetConfiguration("dev")
+	balanceurl := config.EthRPCURL + address
+	resp, err := httpClient.Get(balanceurl)
+
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	type AddressBalance struct {
+		Status  bool     `json:"status"`
+		Message string   `json:"message"`
+		Data    *big.Int `json:"data"`
+	}
+
+	bodyBytes, _ := ioutil.ReadAll(resp.Body)
+	balanceInfo := new(AddressBalance)
+	json.Unmarshal(bodyBytes, balanceInfo)
+
+	if !balanceInfo.Status {
+		return nil, err
```

`err` is nil here, please return an error to indicate what was wrong. Maybe `errors.New("failed to get balance info")`